File: galicia_2022/opcion_a/exercicio2_1.py
def cifra_v1(texto):
    DESPLAZAMIENTO = 5
    alfabeto = ("a","b","c","d","e","f","g","h","i","j","k","l","m",
                "n","o","p","q","r","s","t","u","v","w","x","y","z")
    
    cifrado = ""
    for letra in texto:
        indice_cif = (alfabeto.index(letra) + DESPLAZAMIENTO) % len(alfabeto)
        cifrado += alfabeto[indice_cif]
        
    return cifrado

print(cifra_v1("marcex"))
        

File: galicia_2022/opcion_a/Exercicio2/Ex2_1/Exercicio2_1.java
import java.util.Arrays;
import java.util.List;

public class Exercicio2_1 {
    private static final int DESPLAZAMIENTO = 5;

    public static String cifra(String texto){
        String[] alfabeto = {"a","b","c","d","e","f","g","h","i","j","k","l",
        "m","n","o","p","q","r","s","t","u","v","w","x","y","z"};

        //Lo pasamos a lista porque admiten el método indexOf
        List<String> listaAlfabeto = Arrays.asList(alfabeto);

        String cifrado = "";
        for (char letra : texto.toCharArray()) {
            int indice_cif = (listaAlfabeto.indexOf(String.valueOf(letra)) + DESPLAZAMIENTO) % listaAlfabeto.size();
            cifrado += listaAlfabeto.get(indice_cif);
        }
        return cifrado;
    }
    public static void main(String[] args) {
        //Prueba
        System.out.println(cifra("vicente"));

    }
}